lld/test/COFF/code-merge.s

```diff
@@ -0,0 +1,57 @@
+# REQUIRES: x86
+
+# Test that merging code section into a data section changes its characteristics.
+
+# RUN: llvm-mc -triple x86_64-windows-msvc %s -filetype=obj -o %t.obj
+# RUN: lld-link -machine:amd64 -dll -noentry -out:%t.dll %t.obj -merge:.testx=.testd -merge:.testx2=.testbss
```

Comment:
Add more text/data sections and test the other merge direction?

Reply:
I pushed more tests.

I noticed one more problem. To fill gaps between chunks, we memset the whole section to 0xcc. When merging uninitialized chunks into code section, it means that those chunks will not be zeroed. This is a preexisting problem, not specific to this PR, but this PR makes it a bit more widespread by marking more sections as code sections. I will take a look at that.